edx_proctoring/api.py

```diff
@@ -493,6 +493,26 @@ def remove_allowance_for_user(exam_id, user_id, key):
         emit_event(exam, 'allowance.deleted', override_data=data)
 
 
+def add_bulk_allowances(exam_ids, user_ids, allowance_type, value):
+    """
+    Adds (or updates) an allowance for multiple users and exams
+    """
+    exam_ids = set(exam_ids)
+    user_ids = set(user_ids)
+    if allowance_type == 'time_multiplier':
+        for exam_id in exam_ids:
+            time_allowed = str(round(get_exam_by_id(exam_id)["time_limit_mins"] * (float(value) - 1)))
+            for user_id in user_ids:
+                add_allowance_for_user(exam_id, user_id, 
+                ProctoredExamStudentAllowance.ADDITIONAL_TIME_GRANTED, time_allowed)
+    else:
+        for exam_id in exam_ids:
+            for user_id in user_ids:
+                add_allowance_for_user(exam_id, user_id, ProctoredExamStudentAllowance.ADDITIONAL_TIME_GRANTED, value)
+
+
```

Comment:
Please add logging statements. Do `log.info()` for happy path, and `log.warning()` for edge cases. Use `log.error()` for error cases.

Reply:
For logging statements I'm not too sure exactly what I should be logging as the get allowance function would log when there is an error with a specific allowance is accepted or rejected. Should I log if the allowance value is acceptable?

Reply:
According to the existing [implementation ](https://github.com/edx/edx-proctoring/blob/84df549ba644d7a15556aa1c1281b2ab91a8a524/edx_proctoring/api.py#L430)of `add_allowance_for_user`, it looks like the normal condition is already logged. But I think that's not good enough. You should also log.info the beginning of this function with parameters passed into this function to help future engineers understand this function is invoked. Then you should catch exceptions thrown by `add_allowance_for_user` in this function, log and handle properly by keep going.